eval.go

```diff
@@ -76,23 +77,29 @@ func parseRemoteObject(v *runtime.RemoteObject, res interface{}) (err error) {
 	}
 
 	value := v.Value
-	if value == nil {
-		// if js expression is undefined, it cannot resolve to any go type, so return ErrJSUndefined directly
-		if v.Type == "undefined" {
-			return ErrJSUndefined
-		}
+	if v.Type == "undefined" || value == nil {
 		rv := reflect.ValueOf(res)
+	checkKind:
 		if rv.Kind() == reflect.Pointer {
-			switch rv.Elem().Kind() {
-			case reflect.Interface, reflect.Map, reflect.Pointer, reflect.Slice:
-				// only interface, map, ptr, slice type can resolve js null
-				value = []byte("null")
+			rv = rv.Elem()
+			switch rv.Kind() {
+			case reflect.Map, reflect.Pointer, reflect.Slice,
+				reflect.Func, reflect.Chan, reflect.Struct, reflect.Array:
+				// map, ptr, slice, func, chan, struct, array type can be handled correctly by json package
+			case reflect.Interface:
+				// if `res` is interface, get its actual type and check again
+				rv = rv.Elem()
+				goto checkKind
```

Comment:
Can we have a test case for this code path?

Reply:
yes, test at eval_test.goL244

Reply:
That test does not show why we need

```go
rv = rv.Elem()
goto checkKind
```

Reply:
query_test.go `TestValueUndefined` will be executed to this case

Reply:
func `Value` calls the func `JavascriptAttribute` the parameter res is converted from type * string to interface {}, so we need the above code

Reply:
I see. That's a bug in `callFunctionOnNode`. I have just sent #1264 to fix it. Thank you!

I think we don't need to treat `reflect.Interface` differently now. What do you think?

Reply:
Yes, not now